### PR TITLE
Tw-130: Baker rewards

### PR DESCRIPTION
**Pull request by LendiHop**

### CODE REVIEW

File: src/screens/delegation-screen/selected-baker-screen/baker-rewards-list/baker-reward-item/baker-reward-item-details/baker-reward-item-details.tsx [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+    missedEndorsementRewards,
+    missedOwnBlockFees,
+    missedOwnBlockRewards,
+    missedOwnBlocks,
+    missedEndorsements
+  } = reward;
+
+  return (
+    <>
+      {ownBlocks > 0 && (
```
Comment (letier3110) [inline src/screens/delegation-screen/selected-baker-screen/baker-rewards-list/baker-reward-item/baker-reward-item-details/baker-reward-item-details.tsx:35]: `<OwnBlocks ownBlockRewards={ownBlockRewards} ownBlocks={ownBlocks} ownBlockFees={ownBlockFees}  />`
or even make special prop-object with `{ ownBlockFees, ownBlockRewards, ownBlocks }`

File: src/screens/delegation-screen/selected-baker-screen/baker-rewards-list/baker-reward-item/baker-reward-item-details/baker-reward-item-details.tsx [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+                    <Text style={styles.textGray}> fees</Text>
+                  </Text>
+                </View>
+              </View>
+            </View>
+          </View>
+        </>
+      )}
+
+      {endorsements > 0 && (
```
Comment (letier3110) [inline src/screens/delegation-screen/selected-baker-screen/baker-rewards-list/baker-reward-item/baker-reward-item-details/baker-reward-item-details.tsx:76]: `<Endorsements />`

File: src/screens/delegation-screen/selected-baker-screen/baker-rewards-list/baker-reward-item/baker-reward-item-details/baker-reward-item-details.tsx [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+                    </Text>
+                  </Text>
+                </View>
+              </View>
+            </View>
+          </View>
+        </>
+      )}
+
+      {missedOwnBlocks > 0 && (
```
Comment (letier3110) [inline src/screens/delegation-screen/selected-baker-screen/baker-rewards-list/baker-reward-item/baker-reward-item-details/baker-reward-item-details.tsx:111]: `<MissedOwnBlocks />`

File: src/screens/delegation-screen/selected-baker-screen/baker-rewards-list/baker-reward-item/baker-reward-item-details/baker-reward-item-details.tsx [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+                    <Text style={styles.textGray}> fees</Text>
+                  </Text>
+                </View>
+              </View>
+            </View>
+          </View>
+        </>
+      )}
+
+      {missedEndorsements > 0 && (
```
Comment (letier3110) [inline src/screens/delegation-screen/selected-baker-screen/baker-rewards-list/baker-reward-item/baker-reward-item-details/baker-reward-item-details.tsx:155]: `<MissedEndorsements missedEndorsementRewards={missedEndorsementRewards} missedEndorsements={missedEndorsements} />`

File: src/screens/delegation-screen/selected-baker-screen/baker-rewards-list/baker-reward-item/baker-reward-item.styles.ts [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+    alignItems: 'center'
+  },
+  cellContainer: {
+    marginTop: formatSize(8)
+  },
+  cellTitle: {
+    ...typography.caption11Regular,
+    color: colors.gray1
+  },
+  cellValueText: {
```
Comment (letier3110) [inline src/screens/delegation-screen/selected-baker-screen/baker-rewards-list/baker-reward-item/baker-reward-item.styles.ts:35]: same as `textBlack`

File: src/screens/delegation-screen/selected-baker-screen/baker-rewards-list/baker-reward-item/baker-reward-item.styles.ts [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+    ...typography.numbersRegular15,
+    color: colors.black
+  },
+  detailsButtonText: {
+    ...typography.caption13Semibold,
+    color: colors.destructive,
+    marginRight: formatSize(6)
+  },
+  detailTitleContainer: {
```
Comment (letier3110) [inline src/screens/delegation-screen/selected-baker-screen/baker-rewards-list/baker-reward-item/baker-reward-item.styles.ts:44]: same as `cycleInfoContainer`

File: src/screens/delegation-screen/selected-baker-screen/baker-rewards-list/utils/get-baking-efficiency.ts [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+        endorses: (rewardStruct & 2) > 0,
+        fees: (rewardStruct & 4) > 0,
+        accusationRewards: (rewardStruct & 8) > 0,
+        accusationLostDeposits: (rewardStruct & 16) > 0,
+        accusationLostRewards: (rewardStruct & 32) > 0,
+        accusationLostFees: (rewardStruct & 64) > 0,
+        revelationRewards: (rewardStruct & 128) > 0,
+        revelationLostRewards: (rewardStruct & 256) > 0,
+        revelationLostFees: (rewardStruct & 512) > 0,
+        missedBlocks: (rewardStruct & 1024) > 0,
```
Comment (unixvb) [inline src/screens/delegation-screen/selected-baker-screen/baker-rewards-list/utils/get-baking-efficiency.ts:65]: fix warnings

**Review by letier3110 — APPROVED** [review]

File: src/screens/delegation-screen/selected-baker-screen/baker-rewards-list/utils/reduce-function.ts [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+    const x0 = leftValueIndex;
+    const y0 = leftValue;
+    const x1 = rightValueIndex;
+    const y1 = rightValue;
+    fallbackRewardsValue = new BigNumber(index - x0)
+      .div(x1 - x0)
+      .multipliedBy(y1.minus(y0))
+      .plus(y0);
+  } else if (leftValueExists || rightValueExists) {
+    fallbackRewardsValue = rewardsPerEventHistory[leftValueExists ? leftValueIndex : rightValueIndex][key]!;
```
Comment (unixvb) [inline src/screens/delegation-screen/selected-baker-screen/baker-rewards-list/utils/reduce-function.ts:53]: fix warnings

**Review by unixvb — APPROVED** [review]

**Pull request merged**